label_trans_selinux.go

```diff
@@ -20,6 +18,28 @@ const timeout = time.Second * 3
 
 const nullChar = "\000"
 
+/* This is a cursed function and I hate it.
+ * The Go devlopers have decided that machine byteorder is an unimportant detail
+ * and 1) force you to specify order when decoding byte arrays, and
+ * 2) don't give you a simple way of figuring out the native endianness
+ * so here we go. Create a pointer and then look at the byte order to set
+ * the interface for reading off of the socket. */
+var nativeEndian binary.ByteOrder
+
+func setNativeEndian() {
+	buf := [2]byte{}
+	*(*uint16)(unsafe.Pointer(&buf[0])) = uint16(0xABCD)
+
+	switch buf {
+	case [2]byte{0xCD, 0xAB}:
+		nativeEndian = binary.LittleEndian
+	case [2]byte{0xAB, 0xCD}:
+		nativeEndian = binary.BigEndian
+	default:
+		panic("Could not determine native endianness.")
```

Comment:
Let's make this function return an error instead. It's not good practice to `panic()`